doc/flame/rendering/particles.md

```diff
@@ -1,21 +1,21 @@
 # Particles
 
 Flame offers a basic, yet robust and extendable particle system. The core concept of this system is
-the `Particle` class, which is very similar in its behavior to the `ParticleComponent`.
+the `Particle` class, which is very similar in its behavior to the `ParticleSystemComponent`.
 
 The most basic usage of a `Particle` with `FlameGame` would look as following:
 
 ```dart
 import 'package:flame/components.dart';
 
-// ... 
+// ...
 
 game.add(
-  // Wrapping a Particle with ParticleComponent
+  // Wrapping a Particle with ParticleSystemComponent
   // which maps Component lifecycle hooks to Particle ones
   // and embeds a trigger for removing the component.
-  ParticleComponent(
-    CircleParticle(),
+  ParticleSystemComponent(
+    particle: CircleParticle(),
```

Comment:
This carries a bit of the confusing part in the other component, maybe this could be called `particles`?

Reply:
I wonder if that won't cause another type of confusion, `particles` would give me the idea that this would be a list, and since how our particle system is implemented, a `Particle` instance can hold multiple particles, so maybe naming this `particle` is fine? Wdyt?

Reply:
True, `particleType` or something maybe?

Reply:
`particleType` sounds a little bit weird, I would expect it be an enum or something, but this is the actually value.

Maybe `value`? It maybe a little too generic but maybe better?

Reply:
True, what about `particleGenerator` then? 😄
![CMS-CERN-image-large-635x394.jpg](https://user-images.githubusercontent.com/744771/160132330-b9a90eb8-92fb-45cc-a174-26e230602f1a.jpg)

Reply:
That crossed my mind, but that will in turn gives me the idea that this would be a function hahahaha

Also that is a particle accelerator right? :P

Reply:
> Also that is a particle accelerator right? :P

The days that it is lucky it is both. 😄

Hmm... `particleSpawner`, `spawn` or something like that then?

Reply:
`particleSpawner` I liked the most so far

Reply:
generator do bring me the strong idea that this would be a function even more because `Particle` class has a generate factory that receives a named parameter called generator.

So spawmer would be better here IMO, leaving unnamed would make this not follow the standards of the other components, so my vote goes to spawmer or particleSpawmer

Reply:
but it doesn't spawn anything

Reply:
it does 🤔 , it is that class that spawn the particles

Reply:
Umm, it is a `Particle` object; usually a `CompositeParticle` so that it has many other particles inside. But it is a fully-made particle system.

Of course, it is possible to create a `Particle` that would self-mutate over time and spawn other particles inside, but we don't have any such functionality implemented yet.

Reply:
>  it is a Particle object

Yeah, that is why it was originally named `particle`, maybe we should just stick to that...

Reply:
> it was originally named particle, maybe we should just stick to that...

I would prefer `particle` over `particleSpawner` or `particleGenerator`. However, maybe we could revisit the rule that all arguments must be named, and allow an exception for the case when the meaning of the argument is unambiguous from its type. So that we could write `GameWidget(game)` instead of `GameWidget(game: game)`, and `SpriteComponent(sprite)` instead of `SpriteComponent(sprite: sprite)`, and `ParticleSystemComponent(particle)` instead of `ParticleSystemComponent(particle: particle)`.

Reply:
Yeah, sounds like the best option!